src/Samples/Sample.AspNetCore/Controllers/CheckOutController.cs

```diff
@@ -114,6 +122,41 @@ public async Task<CardPayment> CreateCardPayment()
                 return null;
             }
         }
+        
+
+
+        public async Task<TrustlyPayment> CreateTrustlyPayment()
+        {
+            var totalAmount = this.cartService.CalculateTotal();
+            var vatAmount = Amount.FromDecimal(0);
+            try
+            {
+                var trustlyPaymentRequest = new TrustlyPaymentRequest(new CurrencyCode("SEK"),
+                                                                                     new List<Price>
+                                                                                     {
+                                                                                         new Price(Amount.FromDecimal(totalAmount),
+                                                                                                   PriceType.Trustly, vatAmount)
+                                                                                     },
+                                                                                     "Test Purchase", this.payeeInfoOptions.PayeeReference, "useragent", CultureInfo.GetCultureInfo("sv-SE"),
+                                                                                     new Urls(this.urls.HostUrls, this.urls.CompleteUrl,
+                                                                                              this.urls.TermsOfServiceUrl, this.urls.CancelUrl,
+                                                                                              this.urls.PaymentUrl, this.urls.CallbackUrl, this.urls.LogoUrl),
+                                                                                     new PayeeInfo(this.payeeInfoOptions.PayeeId,
+                                                                                                   this.payeeInfoOptions.PayeeReference), new SwedbankPay.Sdk.Payments.TrustlyPayments.PrefillInfo("Ola", "Nordmann"));
+                var trustlyPayment = await this.swedbankPayClient.Payments.TrustlyPayments.Create(trustlyPaymentRequest);
+                this.cartService.PaymentLink = trustlyPayment.PaymentResponse.Id.OriginalString;
+                this.cartService.Instrument = PaymentInstrument.Trustly;
+                this.cartService.PaymentOrderLink = null;
+                this.cartService.Update();
+
+                return trustlyPayment;
+            }
+            catch (Exception ex)
+            {
+                Debug.Write(ex.Message);
+                return null;
+            }
+        }
 
         public async Task<SwishPayment> CreateSwishPayment()
```

Comment:
This should either be renamed, or the creation of Trustly payments should be moved to its own method.

Reply:
I'm not sure what you're meaning here? It is a own method `CreateTrustlyPayment`

Reply:
Ah my bad. The GitHub UI showed the closest method name as `CreateSwishPayment` for the code below, but it's in the `GetPaymentJsSource`. My bad.